### PR TITLE
feat: enable updating host labels via the control interface

### PR DESCRIPTION
**Pull request by connorsmith256**
## Feature or Problem
This adds two new topics to the host's control interface: `wasmbus.ctl.{lattice_id}.labels.{host_id}.{put/del}`. Publishing on these topics allows users to dynamically adjust the target host's labels

## Related Issues
Relates to https://github.com/wasmCloud/wasmCloud/issues/692

## Release Information
Minor bump to the control interface crate as well as the host. Functionality via wash to be added after both of the former are released

## Consumer Impact
This is a potentially a step in the direction of enabling lame-duck mode on the host (we could use a special label to convey that intent)

### Acceptance or Integration
I exercised put and delete via the wasmbus tests

### Manual Verification
```
wash get hosts -o json | jq -r '.hosts | .[] | .labels'
{
  "hostcore.arch": "aarch64",
  "hostcore.os": "macos",
  "hostcore.osfamily": "unix"
}

nats req wasmbus.ctl.default.labels.NARZ43WTEW56DC3Q42FXIND2WJV5TFVTBYJBJ2T42S5OB2AHIPITDDWW.put '{"label": "foo", "value": "bar"}'

18:31:08 Sending request on "wasmbus.ctl.default.labels.NARZ43WTEW56DC3Q42FXIND2WJV5TFVTBYJBJ2T42S5OB2AHIPITDDWW.put"
18:31:08 Received with rtt 501.459µs
{"accepted":true,"error":""}

wash get hosts -o json | jq -r '.hosts | .[] | .labels'
{
  "foo": "bar",
  "hostcore.arch": "aarch64",
  "hostcore.os": "macos",
  "hostcore.osfamily": "unix"
}

nats req wasmbus.ctl.default.labels.NARZ43WTEW56DC3Q42FXIND2WJV5TFVTBYJBJ2T42S5OB2AHIPITDDWW.del '{"label": "foo"}'

18:31:20 Sending request on "wasmbus.ctl.default.labels.NARZ43WTEW56DC3Q42FXIND2WJV5TFVTBYJBJ2T42S5OB2AHIPITDDWW.del"
18:31:20 Received with rtt 551.917µs
{"accepted":true,"error":""}

wash get hosts -o json | jq -r '.hosts | .[] | .labels'
{
  "hostcore.arch": "aarch64",
  "hostcore.os": "macos",
  "hostcore.osfamily": "unix"
}
```

### CODE REVIEW

**Review by rvolosatovs — APPROVED** [review]

File: crates/control-interface/src/lib.rs [hunk 1]
```diff
@@ -503,6 +503,47 @@ impl Client {
...
+    }
+
+    /// Removes a label from the given host.
+    ///
+    /// # Errors
+    ///
+    /// Will return an error if there is a communication problem with the host
+    pub async fn delete_label(&self, host_id: &str, label: &str) -> Result<CtlOperationAck> {
+        let subject = broker::delete_label(&self.topic_prefix, &self.lattice_prefix, host_id);
+        debug!(%subject, "removing label");
```
Comment (vados-cosmonic) [inline crates/control-interface/src/lib.rs:536]: Hey could you add the `host_id` here? Maybe something like `removing label [{}] from host [{}]`?

File: crates/control-interface/src/lib.rs [hunk 2]
```diff
@@ -503,6 +503,47 @@ impl Client {
...
+        let bytes = json_serialize(HostLabel {
+            label: label.to_string(),
+            value: String::new(), // value isn't parsed by the host
+        })?;
+        match self.request_timeout(subject, bytes, self.timeout).await {
+            Ok(msg) => Ok(json_deserialize(&msg.payload)?),
+            Err(e) => Err(format!("Did not receive remove label acknowledgement: {e}").into()),
```
Comment (vados-cosmonic) [inline crates/control-interface/src/lib.rs:543]: NIT: should this be capital (control-interface is internal right?), and could we `context` here?

File: crates/host/src/wasmbus/mod.rs [hunk 3]
```diff
@@ -3675,7 +3695,7 @@ impl Host {
 
     #[instrument(level = "trace", skip_all)]
     async fn handle_config_get_one(&self, entity_id: &str, key: &str) -> anyhow::Result<Bytes> {
-        trace!("getting config");
+        trace!("handling config");
```
Comment (vados-cosmonic) [inline crates/host/src/wasmbus/mod.rs:3698]:
```suggestion
        trace!(%entity_id, "handling config");
```
To match the other one?

File: crates/control-interface/src/types.rs [hunk 4]
```diff
@@ -418,3 +418,9 @@ pub struct LinkDefinition {
     pub contract_id: String,
     pub values: LinkSettings,
 }
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct HostLabel {
+    pub label: String,
```
Comment (brooksmtownsend) [inline crates/control-interface/src/types.rs:424]: I'm going back and forth on this being `label` or `key`, did you consider `key` so it's just submitting kv pairs?

**Review by brooksmtownsend — APPROVED** [review]
I'm happy with this, and I think the topics fit into our current structure. We may want to change them when we audit the control interface topics but that's okay

File: crates/control-interface/src/types.rs (hunk 4 above)
Comment (connorsmith256) [reply crates/control-interface/src/types.rs:424]: Hmm I didn't think about this, but I like what you're saying. A "label" is really a key + value

I'll rename 👍

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

File: crates/control-interface/src/lib.rs (hunk 2 above)
Comment (vados-cosmonic) [reply crates/control-interface/src/lib.rs:543]: Ah thanks for clarifying this -- the output doesn't look too unreasonable

File: crates/control-interface/src/lib.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/control-interface/src/lib.rs:536]: hmnnn I know that as someone debugging something perhaps I'd really want to know *what* the label was removed from, but definitely shouldn't hold up this PR -- we'll cross that :bridge_at_night: if we get there

**Pull request merged**